ddtrace/internal/datastreams/kombu.py

```diff
@@ -0,0 +1,46 @@
+from ddtrace import config
+from ddtrace.contrib.kombu.utils import HEADER_POS
+from ddtrace.contrib.kombu.utils import PUBLISH_BODY_IDX
+from ddtrace.contrib.kombu.utils import get_routing_key_from_args
+from ddtrace.contrib.kombu.utils import get_exchange_from_args
+from ddtrace.internal import core
+from ddtrace.internal.datastreams.processor import PROPAGATION_KEY
+from ddtrace.internal.datastreams.utils import _calculate_byte_size
+from ddtrace.internal.logger import get_logger
+
+
+log = get_logger(__name__)
+
+
+def handle_kombu_produce(args, kwargs, span):
+    from . import data_streams_processor as processor
+
+    dsm_identifier = get_exchange_from_args(args)
+    payload_size = 0
+    payload_size += _calculate_byte_size(args[HEADER_POS])
+    payload_size += _calculate_byte_size(args[PUBLISH_BODY_IDX])
+
+    has_routing_key = str(bool(get_routing_key_from_args(args))).lower()
+
+    pathway = processor().set_checkpoint(
+        ["direction:out", f"exchange:{dsm_identifier}", f"has_routing_key:{has_routing_key}", "type:rabbitmq"], payload_size=payload_size, span=span
```

Comment:
sometimes, the exchange is unset, does the logic work in this case?

Reply:
What is the expected value when unset, right now it will result as "None"